boards/arm/nrf54l15dk_nrf54l15/revision.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# This is a list of all possible revisions
+set(revisions "soc1")
+
+# Ensure that the revision set by the user is supported
+if (DEFINED BOARD_REVISION)
+  if (NOT BOARD_REVISION IN_LIST revisions)
+    message(FATAL_ERROR
+      "'${BOARD_REVISION}' is not a valid revision for ${BOARD} Valid revisions are '${revisions}'")
+  endif()
+
+  set(overlay_file ${BOARD_DIR}/${BOARD}_${BOARD_REVISION}.overlay)
```

Comment:
why is this needed ?

Zephyr already have: https://github.com/zephyrproject-rtos/zephyr/blob/18deb582efc9de666c5648ffe7e6068ea319dbf9/cmake/modules/dts.cmake#L128